### ISSUE TITLE
could you given an example of installation on mac? or even a homebrew formula?

### ISSUE THREAD

**Issue by autosquid** [qnzhou/PyMesh#11]
could you given an example of installation on mac? or even a homebrew formula?

**Comment by qnzhou** [comment]
Right now, the only way of installing on mac is the same as on linux.  That is to compile from the source code.  If you are into cloud computing, PyMesh is also containerized: https://hub.docker.com/r/qnzhou/pymesh/

I am looking into the option of packaging PyMesh to place it homebrew and macport, but that may take some time.

**Comment by ksanjeevan** [comment]
I'm getting the following error when running `cmake ..`

`-- CMAKE_INSTALL_RPATH: /Users/ksanjeevan/anaconda/lib/python3.5/site-packages/pymesh-0.1-py3.5.egg/pymesh/lib;/Users/ksanjeevan/.local/lib/python3.5/site-packages/pymesh-0.1-py3.5.egg/pymesh/lib;/usr/local/pymesh-0.1-py3.5.egg/pymesh/lib;/Users/ksanjeevan/anaconda/lib/python3.5/site-packages/pymesh-0.1-py3.5.egg/pymesh/third_party/lib;/Users/ksanjeevan/.local/lib/python3.5/site-packages/pymesh-0.1-py3.5.egg/pymesh/third_party/lib;/usr/local/pymesh-0.1-py3.5.egg/pymesh/third_party/lib;/Users/ksanjeevan/PyMesh/python/pymesh/lib;/Users/ksanjeevan/PyMesh/python/pymesh/third_party/lib
-- Boost version: 1.62.0
CMake Error at cmake/FindAllDependencies.cmake:49 (INCLUDE):
  include called with wrong number of arguments.  include() only takes one
  file.
Call Stack (most recent call first):
  Settings.cmake:59 (FIND_PACKAGE)
  CMakeLists.txt:16 (INCLUDE)


-- BSP library cannot be found.  Consider set BSP_PATH environment variable (missing:  BSP_INCLUDE_DIRS BSP_LIBRARIES) 
-- Could NOT find UMFPACK (missing:  UMFPACK_INCLUDES UMFPACK_LIBRARIES) 
-- Could NOT find CHOLMOD (missing:  CHOLMOD_INCLUDES CHOLMOD_LIBRARIES) 
-- METIS library cannot be found.  Consider set METIS_PATH environment variable (missing:  METIS_INCLUDE_DIRS METIS_LIBRARIES) 
-- tinyxml2 library cannot be found.  Consider set TINYXML2_PATH environment variable (missing:  TINYXML2_INCLUDE_DIRS TINYXML2_LIBRARIES) 
CMake Warning at tools/ConvexHull/CMakeLists.txt:17 (MESSAGE):
  CGAL missing.  Skipping CGAL convex hull support


CMake Warning at tools/Boolean/CMakeLists.txt:45 (MESSAGE):
  CGAL is missing, libigl boolean support is skipped


CMake Warning at tools/Boolean/CMakeLists.txt:55 (MESSAGE):
  CGAL is missing, CGAL boolean support is skipped


CMake Warning at tools/SelfIntersection/CMakeLists.txt:20 (MESSAGE):
  CGAL is missing, libigl self-intersection support is skipped


CMake Warning at tools/OuterHull/CMakeLists.txt:20 (MESSAGE):
  CGAL is missing, libigl outer hull support is skipped


CMake Warning at tools/WindingNumber/CMakeLists.txt:20 (MESSAGE):
  CGAL is missing, libigl winding number support is skipped


CMake Warning at tools/CMakeLists.txt:58 (MESSAGE):
  CGAL wrapper module is not built because CGAL lib is missing


CMake Warning at tools/CMakeLists.txt:72 (MESSAGE):
  Tetrahedronization module is not built because CGAL is missing


CMake Warning at tools/CMakeLists.txt:93 (MESSAGE):
  SparseSolver module is not built because Umfpack is missing


CMake Warning at tools/CMakeLists.txt:104 (MESSAGE):
  Wires module is not built because CGAL is missing


-- Configuring incomplete, errors occurred!
See also "/Users/ksanjeevan/PyMesh/build/CMakeFiles/CMakeOutput.log".`

Also, it would be awesome if this was on homebrew! Do you have an ETA on when that might be? Thanks so much.

**Comment by qnzhou** [comment]
Thanks for the report.  One of the dependencies, CGAL, is missing.  Have you installed CGAL on your computer?

I will definitely look into homebrew or macport option, but I cannot give out an estimate timeline right now...  Sorry.

**Comment by ksanjeevan** [comment]
I have cgal-4.9 installed, and I set CGAL_PATH variable to its directory (/usr/local/Cellar/cgal)...

**Comment by qnzhou** [comment]
```$CGAL_PATH``` to be set tot the directory containing ```CGALConfig.cmake```.  Could you check if the following file exists?  ```$CGAL_PATH/CGALConfig.cmake```

**Comment by ksanjeevan** [comment]
Turns out `CGALConfig.cmake` was in `/usr/local/Cellar/cgal/4.9/lib/CGAL`! Now the unit tests have passed but when running the final `python -c "import pymesh; pymesh.test()"` I'm getting an import error: `ImportError: No module named 'PySparseSolver'`. Is `PySparseSolver` compatible with python3? 

Thanks.

**Comment by qnzhou** [comment]
Could you ```git pull``` again to see if it is fixed?

**Comment by ksanjeevan** [comment]
Fixed! Working perfectly. Really appreciate your help with all of this!

**Comment by wenhuizhang** [comment]
Hi, after installation, it was smooth, while testing , I got "'module' object has no attribute 'load_mesh'"

**Comment by qnzhou** [comment]
That is really strange.  So ```import pymesh``` succeeded, but ```pymesh.test()``` failed with error.  Have you previously installed other package that is also called ```pymesh```?  What is the output of ```dir(pymesh)```?

**Comment by smor** [comment]
Hi, I had the same error than @wenhuizhang because I first tried and install it through pip with `pip install pymesh`, but it must be this package instead : https://pypi.python.org/pypi/pymesh/0.1.0

**Comment by wenhuizhang** [comment]
I was wondering if it is due to the deprecated link issue on the backend of website of https://pypi.python.org/pypi/pymesh/0.1.0  ??

**Comment by qnzhou** [comment]
Right, what you installed through pip is a [completely different package](https://github.com/taxpon/pymesh).  Takuro, author of the other package, registered his package before me, so he got the name `pymesh`.
